### PR TITLE
completed task 6.

### PR DESCRIPTION
**Pull request by CodeCanto**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -123,3 +123,29 @@ describe("GET /api/reviews", () => {
...
+      .then(({ body }) => {
+        const { comments } = body;
+        comments.forEach((review) => {
+          expect(review).toMatchObject({
+            comment_id: expect.any(Number),
+            votes: expect.any(Number),
+            created_at: expect.any(String),
+            author: expect.any(String),
+            body: expect.any(String),
+            review_id: expect.any(Number),
```
Comment (Dutton1990) [inline __tests__/app.test.js]: The other values are fine to expect.any, but the review_id should be 1 if that's the endpoint you're going to!

**Review by Dutton1990 — CHANGES_REQUESTED** [review]
Nice job so far with the happy path! Looking great. 

Now to move on to sad paths. 
You can do similar checks to what you've done before for the review_id, so 'not-a-number' and 999 for example, you should check these on this endpoint too. 

There's a trickier test for this endpoint though, if a review exists but has no comments **yet** you want to return an empty array of comments with a 200 status code. The review is valid, there's just no comments yet. 

To do this you'll have to implement something similar to what Dave did in the lecture this morning, by checking if that review exists. 

See if you can implement these tests and re-send this PR to us once you've given it a go!

File: __tests__/app.test.js [hunk 2]
```diff
@@ -123,3 +123,53 @@ describe("GET /api/reviews", () => {
...
+          expect(review).toMatchObject({
+            comment_id: expect.any(Number),
+            votes: expect.any(Number),
+            created_at: expect.any(String),
+            author: expect.any(String),
+            body: expect.any(String),
+            review_id: 2,
+          });
+          const { comments } = body;
+          const sortedComments = [...comments].sort((commentA, commentB) => {
```
Comment (RichardNeat) [inline __tests__/app.test.js:144]: We could make this a bit easier by using jest sorted like we were using last week.

File: models/reviewsModel.js [hunk 3]
```diff
@@ -30,3 +30,26 @@ exports.fetchReviews = () => {
...
+    .then((results) => {
+      if (results.rowCount === 0) {
+        return Promise.reject({ status: 404, msg: "Not found" });
+      }
+      return db.query(`SELECT * FROM comments WHERE review_id = $1`, [
+        reviewId,
+      ]);
+    })
+    .then((results) => {
+      let comments = results.rows;
```
Comment (RichardNeat) [inline models/reviewsModel.js:46]: const over let 😉

File: models/reviewsModel.js [hunk 4]
```diff
@@ -30,3 +30,26 @@ exports.fetchReviews = () => {
...
+      if (comments.length === 0) {
+        return [];
+      }
+      return comments;
```
Comment (RichardNeat) [inline models/reviewsModel.js:50]: you may not need this if statement here as rows will be an empty array when there are no responses so returning comments in any scenario should do the job.

**Review by RichardNeat — COMMENTED** [review]
Really nice James, just a couple of considerations I've left as notes in your code. Have a look at refactoring those lines and then go ahead and merge no need to resubmit. Any questions give me a shout 👍

**Pull request merged**